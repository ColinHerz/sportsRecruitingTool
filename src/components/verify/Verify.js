import apiCall from "../../api/api.js";
import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";

import "./verify.scss";

const Verify = props => {
	const { token } = useParams();

	const [firstLoad, setFirstLoad] = useState(true);
	const [verified, setVerified] = useState(false);
	const [showError, setShowError] = useState(false);

	useEffect(() => {
		if (firstLoad) {
			apiCall(
				{
					endpoint: `/users/verify/${token}`,
					type: `GET`
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					setVerified(true);
				},
				() => {
					setShowError(true);
				}
			);

			setFirstLoad(false);
		}
	}, [firstLoad]);

	return (
		<main id="verify">
			<h2>Email Verification</h2>

			{
				verified ?
					<p>Your email has been verified.</p>:
					<p>Verifying your email...</p>
			}

			{
				showError ?
					<p id="error">Something went wrong. Please try again later.</p>:
					null
			}
		</main>
	);
};

export default Verify;
