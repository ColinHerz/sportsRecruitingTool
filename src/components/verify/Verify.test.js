/* eslint-env jest */

import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

import Verify from "./Verify.js";

const genProps = () => {
	return {};
};

describe(`Rendering`, () => {
	it(`Renders correctly to start`, () => {
		const props = genProps();

		const component = renderer.create(
			<MemoryRouter>
				<Verify {...props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
