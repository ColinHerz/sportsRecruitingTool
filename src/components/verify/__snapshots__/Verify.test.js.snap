// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders correctly to start 1`] = `
<main
  id="verify"
>
  <h2>
    Email Verification
  </h2>
  <p>
    Verifying your email...
  </p>
</main>
`;
